ganglia/init.sh

```diff
@@ -8,16 +8,19 @@ rm -rf /mnt/ganglia/rrds/*
 mkdir -p /mnt/ganglia/rrds
 chown -R nobody:nobody /mnt/ganglia/rrds
 
-# Install ganglia
-# TODO: Remove this once the AMI has ganglia by default
+OLD_GANGLIA_PACKAGES="httpd* php* ganglia* ganglia* ganglia-gmond* ganglia-gmetad*"
+GANGLIA_PACKAGES="httpd24-2.4* php56-5.6* ganglia-3.6* ganglia-web-3.5* ganglia-gmond-3.6* ganglia-gmetad-3.6*"
 
-GANGLIA_PACKAGES="ganglia ganglia-web ganglia-gmond ganglia-gmetad"
+#Uninstalls older version of ganglia from master if it was reinstalled in AMI
+yum remove -q -y $OLD_GANGLIA_PACKAGES & sleep 1
+wait
+yum install -q -y $GANGLIA_PACKAGES & sleep 1
+wait
 
-if ! rpm --quiet -q $GANGLIA_PACKAGES; then
-  yum install -q -y $GANGLIA_PACKAGES;
-fi
 for node in $SLAVES $OTHER_MASTERS; do
-  ssh -t -t $SSH_OPTS root@$node "if ! rpm --quiet -q $GANGLIA_PACKAGES; then yum install -q -y $GANGLIA_PACKAGES; fi" & sleep 0.3
+  #Uninstalls older version of ganglia from other masters if it was reinstalled in AMI
+  ssh -t -t $SSH_OPTS root@$node "yum remove -q -y $OLD_GANGLIA_PACKAGES" & sleep 0.3
```

Comment:
I tried this out today and I ran into error messages of the form `Existing lock /var/run/yum.pid: another copy is running as pid 2758.` 

I think the problem is that while the `yum remove is running` we are also starting the `yum install`. I think a good thing to do here would be to not put the `ssh` in background in the first line. Also we can probably use `pssh` to parallelize these steps rather than have a loop. @nchammas did some work on installing `pssh` on the driver node if I am not wrong